### PR TITLE
Compressed_write_without_json_update

### LINKED ISSUE #919: webknossos new CLI JSON error with multiprocessing futures
## Context
- webknossos CLI version 0.13.0
- cubing tiff series of 5714 images, images require padding, total size 121G
- the same series cubes without error in an older wkcuber version (0.9.21)

## Expected Behavior
- cubing should proceed and create mag1 wkw dataset without errors

## Current Behavior

I ran from command line using this command:
```
webknossos convert --jobs 72 --voxel-size 256,256,35 --compress --name Retina_macaque_MEA-align2-regions /gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Retina_macaque_MEA/meta/webknossos/Retina_macaque_MEA-align2-regions-tiffs /gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Retina_macaque_MEA/meta/webknossos/Retina_macaque_MEA-align2-regions
```

It results in this error and terminates before any cubing occurs:
```
ERROR:root:A future crashed with an exception: Expecting value: line 1 column 1 (char 0). Future: <Future at 0x2b6ccbd4a730 state=finished raised JSONDecodeError>
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│                                                                                                  │
│ /gpfs/soma_fs/home/watkins/mambaforge/envs/wkw-new/lib/python3.9/site-packages/webknossos/cli/co │
│ nvert.py:95 in main                                                                              │
│                                                                                                  │
│    92 │   )                                                                                      │
│    93 │                                                                                          │
│    94 │   with get_executor_for_args(args=executor_args) as executor:                            │
│ ❱  95 │   │   dataset = Dataset.from_images(                                                     │
│    96 │   │   │   source,                                                                        │
│    97 │   │   │   target,                                                                        │
│    98 │   │   │   voxel_size,                                                                    │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │              compress = True                                                                 │ │
│ │           data_format = <DataFormat.WKW: 'wkw'>                                              │ │
│ │ distribution_strategy = <DistributionStrategy.MULTIPROCESSING: 'multiprocessing'>            │ │
│ │              executor = <cluster_tools.executors.multiprocessing.MultiprocessingExecutor     │ │
│ │                         object at 0x2b6cbb79c4c0>                                            │ │
│ │         executor_args = Namespace(jobs=72,                                                   │ │
│ │                         distribution_strategy=<DistributionStrategy.MULTIPROCESSING:         │ │
│ │                         'multiprocessing'>, job_resources=None)                              │ │
│ │         job_resources = None                                                                 │ │
│ │                  jobs = 72                                                                   │ │
│ │                  name = 'Retina_macaque_MEA-align2-regions'                                  │ │
│ │                source = PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Retina_ma… │ │
│ │                target = PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Retina_ma… │ │
│ │            voxel_size = VoxelSize(x=256.0, y=256.0, z=35.0)                                  │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│ /gpfs/soma_fs/home/watkins/mambaforge/envs/wkw-new/lib/python3.9/site-packages/webknossos/datase │
│ t/dataset.py:640 in from_images                                                                  │
│                                                                                                  │
│    637 │   │   │   │   )                                                                         │
│    638 │   │   │   else:                                                                         │
│    639 │   │   │   │   category = layer_category                                                 │
│ ❱  640 │   │   │   ds.add_layer_from_images(                                                     │
│    641 │   │   │   │   filepaths[0] if len(filepaths) == 1 else filepaths,                       │
│    642 │   │   │   │   layer_name,                                                               │
│    643 │   │   │   │   category=category,                                                        │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │                 batch_size = None                                                            │ │
│ │                   category = 'color'                                                         │ │
│ │                chunk_shape = None                                                            │ │
│ │           chunks_per_shard = None                                                            │ │
│ │                        cls = <class 'webknossos.dataset.dataset.Dataset'>                    │ │
│ │                   compress = False                                                           │ │
│ │                data_format = <DataFormat.WKW: 'wkw'>                                         │ │
│ │                         ds = Dataset(PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/brigg… │ │
│ │                   executor = <cluster_tools.executors.multiprocessing.MultiprocessingExecut… │ │
│ │                              object at 0x2b6cbb79c4c0>                                       │ │
│ │                  filepaths = [                                                               │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   ... +5704                                                   │ │
│ │                              ]                                                               │ │
│ │        filepaths_per_layer = {                                                               │ │
│ │                              │   'Retina_macaque_MEA-align2-regions-tiffs': [                │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │                                                           │ │
│ │                              PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                              │   │   ... +5704                                               │ │
│ │                              │   ]                                                           │ │
│ │                              }                                                               │ │
│ │                     flip_x = False                                                           │ │
│ │                     flip_y = False                                                           │ │
│ │                     flip_z = False                                                           │ │
│ │    get_valid_pims_suffixes = <function get_valid_pims_suffixes at 0x2b6ccbbbd3a0>            │ │
│ │                 input_file = PosixPath('wafer03_manifest01329_1330_S1330R1330_stitched.tiff… │ │
│ │                input_files = [                                                               │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer03_manifest00684_0685_S685R685_stitched.tiff'), │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer02_manifest00371_0372_S372R372_stitched.tiff'), │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer01_manifest01034_1035_S1035R1035_stitched.tiff… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer01_manifest01487_1488_S1488R1488_stitched.tiff… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer03_manifest00411_0412_S412R412_stitched.tiff'), │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer01_manifest01511_1512_S1512R1512_stitched.tiff… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer02_manifest01139_1140_S1140R1140_stitched.tiff… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer01_manifest00815_0816_S816R816_stitched.tiff'), │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer03_manifest01307_1308_S1308R1308_stitched.tiff… │ │
│ │                              │                                                               │ │
│ │                              PosixPath('wafer03_manifest00894_0895_S895R895_stitched.tiff'), │ │
│ │                              │   ... +5704                                                   │ │
│ │                              ]                                                               │ │
│ │                 input_path = PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                input_upath = PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │             layer_category = None                                                            │ │
│ │                 layer_name = 'Retina_macaque_MEA-align2-regions-tiffs'                       │ │
│ │ map_filepath_to_layer_name = <function                                                       │ │
│ │                              Dataset.ConversionLayerMapping._to_callable.<locals>.<lambda>   │ │
│ │                              at 0x2b6ccbbbd670>                                              │ │
│ │                 max_layers = 20                                                              │ │
│ │                       name = 'Retina_macaque_MEA-align2-regions'                             │ │
│ │                output_path = PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Reti… │ │
│ │                    swap_xy = False                                                           │ │
│ │             use_bioformats = None                                                            │ │
│ │             valid_suffixes = {                                                               │ │
│ │                              │   'gdal',                                                     │ │
│ │                              │   'jpeg',                                                     │ │
│ │                              │   'gif',                                                      │ │
│ │                              │   'ppmraw',                                                   │ │
│ │                              │   'pgm',                                                      │ │
│ │                              │   'ics',                                                      │ │
│ │                              │   'ipl',                                                      │ │
│ │                              │   'stk',                                                      │ │
│ │                              │   'mp4',                                                      │ │
│ │                              │   'avi',                                                      │ │
│ │                              │   ... +59                                                     │ │
│ │                              }                                                               │ │
│ │                 voxel_size = VoxelSize(x=256.0, y=256.0, z=35.0)                             │ │
│ │          z_slices_sort_key = functools.partial(<function natsort_key at 0x2b6e9d47cd30>,     │ │
│ │                              key=None, string_func=<function                                 │ │
│ │                              parse_string_factory.<locals>.func at 0x2b6f26e68310>,          │ │
│ │                              bytes_func=<function parse_bytes_factory.<locals>.<lambda> at   │ │
│ │                              0x2b6f26e683a0>, num_func=<function                             │ │
│ │                              parse_number_factory.<locals>.func at 0x2b6f26e68430>)          │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /gpfs/soma_fs/home/watkins/mambaforge/envs/wkw-new/lib/python3.9/site-packages/webknossos/datase │
│ t/dataset.py:1234 in add_layer_from_images                                                       │
│                                                                                                  │
│   1231 │   │   │   │   │   module="webknossos",                                                  │
│   1232 │   │   │   │   )                                                                         │
│   1233 │   │   │   │   with get_executor_for_args(None, executor) as executor:                   │
│ ❱ 1234 │   │   │   │   │   shapes_and_max_ids = wait_and_ensure_success(                         │
│   1235 │   │   │   │   │   │   executor.map_to_futures(func_per_chunk, args),                    │
│   1236 │   │   │   │   │   │   progress_desc="Creating layer from images",                       │
│   1237 │   │   │   │   │   )                                                                     │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │                                      _ = 0                                                   │ │
│ │                       add_layer_kwargs = {}                                                  │ │
│ │                  allow_multiple_layers = True                                                │ │
│ │                                   args = [                                                   │ │
│ │                                          │   (0, 32),                                        │ │
│ │                                          │   (32, 64),                                       │ │
│ │                                          │   (64, 96),                                       │ │
│ │                                          │   (96, 128),                                      │ │
│ │                                          │   (128, 160),                                     │ │
│ │                                          │   (160, 192),                                     │ │
│ │                                          │   (192, 224),                                     │ │
│ │                                          │   (224, 256),                                     │ │
│ │                                          │   (256, 288),                                     │ │
│ │                                          │   (288, 320),                                     │ │
│ │                                          │   ... +169                                        │ │
│ │                                          ]                                                   │ │
│ │                             batch_size = 32                                                  │ │
│ │                               category = 'color'                                             │ │
│ │                                channel = None                                                │ │
│ │                            chunk_shape = None                                                │ │
│ │                             chunk_size = None                                                │ │
│ │                       chunks_per_shard = None                                                │ │
│ │                               compress = False                                               │ │
│ │                          current_dtype = dtype('uint8')                                      │ │
│ │                            czi_channel = None                                                │ │
│ │                            data_format = <DataFormat.WKW: 'wkw'>                             │ │
│ │                            dimwise_max = <function dimwise_max at 0x2b6cbb7b83a0>            │ │
│ │                                  dtype = None                                                │ │
│ │                               executor = <cluster_tools.executors.multiprocessing.Multiproc… │ │
│ │                                          object at 0x2b6cbb79c4c0>                           │ │
│ │                            first_layer = None                                                │ │
│ │                                 flip_x = False                                               │ │
│ │                                 flip_y = False                                               │ │
│ │                                 flip_z = False                                               │ │
│ │                         func_per_chunk = functools.partial(<bound method                     │ │
│ │                                          PimsImages.copy_to_view of                          │ │
│ │                                          <webknossos.dataset._utils.pims_images.PimsImages   │ │
│ │                                          object at 0x2b6cab443e50>>,                         │ │
│ │                                          mag_view=MagView(name='1',                          │ │
│ │                                          bounding_box=BoundingBox(topleft=(0, 0, 0),         │ │
│ │                                          size=(4956, 5459, 5714))), is_segmentation=False,   │ │
│ │                                          dtype=dtype('uint8'))                               │ │
│ │                                 images = [                                                   │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │                                                   │ │
│ │                                          PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/b… │ │
│ │                                          │   ... +5704                                       │ │
│ │                                          ]                                                   │ │
│ │                                  layer = Layer('Retina_macaque_MEA-align2-regions-tiffs',    │ │
│ │                                          dtype_per_channel=uint8, num_channels=1)            │ │
│ │                             layer_name = 'Retina_macaque_MEA-align2-regions-tiffs'           │ │
│ │                      layer_name_suffix = ''                                                  │ │
│ │                                    mag = Mag(_mag=Vec3Int(1,1,1))                            │ │
│ │                               mag_view = MagView(name='1',                                   │ │
│ │                                          bounding_box=BoundingBox(topleft=(0, 0, 0),         │ │
│ │                                          size=(4956, 5459, 5714)))                           │ │
│ │                             max_layers = 20                                                  │ │
│ │                            pims_images = <webknossos.dataset._utils.pims_images.PimsImages   │ │
│ │                                          object at 0x2b6cab443e50>                           │ │
│ │                       pims_open_kwargs = {}                                                  │ │
│ │                             PimsImages = <class                                              │ │
│ │                                          'webknossos.dataset._utils.pims_images.PimsImages'> │ │
│ │                        possible_layers = None                                                │ │
│ │                                   self = Dataset(PosixPath('/gpfs/soma_fs/cne-mSEM/mSEM-pro… │ │
│ │ suffix_with_pims_open_kwargs_per_layer = {'': {}}                                            │ │
│ │                                swap_xy = False                                               │ │
│ │                              timepoint = None                                                │ │
│ │                                topleft = Vec3Int(0,0,0)                                      │ │
│ │                   truncate_rgba_to_rgb = True                                                │ │
│ │                         use_bioformats = None                                                │ │
│ │                                  z_end = 5714                                                │ │
│ │                                z_start = 5696                                                │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ /gpfs/soma_fs/home/watkins/mambaforge/envs/wkw-new/lib/python3.9/site-packages/webknossos/utils. │
│ py:137 in wait_and_ensure_success                                                                │
│                                                                                                  │
│   134 │   │   with get_rich_progress() as progress:                                              │
│   135 │   │   │   task = progress.add_task(progress_desc, total=len(futures))                    │
│   136 │   │   │   for fut in as_completed(futures):                                              │
│ ❱ 137 │   │   │   │   results.append(fut.result())                                               │
│   138 │   │   │   │   progress.update(task, advance=1)                                           │
│   139 │   return results                                                                         │
│   140                                                                                            │
│                                                                                                  │
│ ╭───────────────────────────────────── locals ─────────────────────────────────────╮             │
│ │           fut = <Future at 0x2b6ccbd01730 state=finished raised JSONDecodeError> │             │
│ │       futures = [                                                                │             │
│ │                 │   <Future at 0x2b6cd3e123d0 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6ccbbda1f0 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3e12460 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3e12910 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3e12be0 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3e129a0 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3eb8e50 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3eb8370 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3eb8f70 state=finished returned tuple>,    │             │
│ │                 │   <Future at 0x2b6cd3eb8c40 state=finished returned tuple>,    │             │
│ │                 │   ... +169                                                     │             │
│ │                 ]                                                                │             │
│ │      progress = <rich.progress.Progress object at 0x2b6cd3d5d850>                │             │
│ │ progress_desc = 'Creating layer from images'                                     │             │
│ │       results = []                                                               │             │
│ │          task = 0                                                                │             │
│ ╰──────────────────────────────────────────────────────────────────────────────────╯             │
│                                                                                                  │
│ /gpfs/soma_fs/home/watkins/mambaforge/envs/wkw-new/lib/python3.9/concurrent/futures/_base.py:439 │
│ in result                                                                                        │
│                                                                                                  │
│   436 │   │   │   │   if self._state in [CANCELLED, CANCELLED_AND_NOTIFIED]:                     │
│   437 │   │   │   │   │   raise CancelledError()                                                 │
│   438 │   │   │   │   elif self._state == FINISHED:                                              │
│ ❱ 439 │   │   │   │   │   return self.__get_result()                                             │
│   440 │   │   │   │                                                                              │
│   441 │   │   │   │   self._condition.wait(timeout)                                              │
│   442                                                                                            │
│                                                                                                  │
│ ╭──── locals ────╮                                                                               │
│ │    self = None │                                                                               │
│ │ timeout = None │                                                                               │
│ ╰────────────────╯                                                                               │
│                                                                                                  │
│ /gpfs/soma_fs/home/watkins/mambaforge/envs/wkw-new/lib/python3.9/concurrent/futures/_base.py:391 │
│ in __get_result                                                                                  │
│                                                                                                  │
│   388 │   def __get_result(self):                                                                │
│   389 │   │   if self._exception:                                                                │
│   390 │   │   │   try:                                                                           │
│ ❱ 391 │   │   │   │   raise self._exception                                                      │
│   392 │   │   │   finally:                                                                       │
│   393 │   │   │   │   # Break a reference cycle with the exception in self._exception            │
│   394 │   │   │   │   self = None                                                                │
│                                                                                                  │
│ ╭── locals ───╮                                                                                  │
│ │ self = None │                                                                                  │
│ ╰─────────────╯                                                                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

I ran the cubing to completion using an older version of the wkcuber CLI (0.9.21), which I modified so that it supports the  `--pad` argument with this command line:
```
python -m wkcuber --jobs 72 --scale 256,256,35 --max_mag 32 --sampling_mode constant_z --pad --name Retina_macaque_MEA-align2-regions /gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Retina_macaque_MEA/meta/webknossos/Retina_macaque_MEA-align2-regions-tiffs /gpfs/soma_fs/cne-mSEM/mSEM-proc/2021/briggman/Retina_macaque_MEA/meta/webknossos/Retina_macaque_MEA-align2-regions
```

## Steps to Reproduce the bug

It's not clear to me the source of the issue, but unless it's something specific about my environment or some specific combination of options, it seems to me that just running `webknossos convert` using multiprocessing on a tiff series should reproduce the error.

## Your Environment for bug

```
(wkw-new) [watkins@somalogin01 cubing]$ python --version
Python 3.9.16
(wkw-new) [watkins@somalogin01 cubing]$ pip list installed | grep webknossos
webknossos                0.13.0
(wkw-new) [watkins@somalogin01 cubing]$ uname -a
Linux somalogin01 3.10.0-1160.el7.x86_64 #1 SMP Mon Oct 19 16:18:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

webknossos CLI was installed with:
```
pip install "webknossos[all]"
```

### PR DESCRIPTION
**Pull request by markbader**
### Description:
- Fixes a bug where the avoidance of bbox updates leads to an AssertionError in `View._prepare_compressed_write`. 

### Issues:
- fixes #919

### Todos:
Make sure to delete unnecessary points or to check all before merging:
 - [x] Updated Changelog
 - [x] Added / Updated Tests

### CODE REVIEW

File: webknossos/Changelog.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ For upgrade instructions, please check the respective _Breaking Changes_ section
 ### Changed
 
 ### Fixed
+- Fixed a bug where the avoidance of bbox updates leads to an AssertionError in `View._prepare_compressed_write`.
```
Comment (normanrz) [inline webknossos/Changelog.md:22]:
```suggestion
- Fixed a bug in writing compressed data. [#942](https://github.com/scalableminds/webknossos-libs/pull/942)
```

File: webknossos/testdata/tiff_with_different_dimensions/1.tif
Comment (normanrz) [inline webknossos/testdata/tiff_with_different_dimensions/1.tif:1]: maybe `tiff_with_different_shapes`

**Review by normanrz — APPROVED** [review]

**Comment by markbader** [comment]
I changed the `_prepare_compressed_write` method to show the warning, but only when the json update is allowed. Since the correct size of the bounding box is unknown, I believe that the error message cannot be reliably output either.

**Review by normanrz — APPROVED** [review]
Good stuff!

**Pull request merged**